src/service/repo/anonymousMessageRepo.ts

```diff
@@ -38,11 +52,12 @@ async function checkIsTargetAllowingAnonDM(targetUserId: string) {
   }
 }
 
-async function sendAnonymousMessage(channelId: string, message: string) {
-  const payload = {
-    channelId,
-    message
-  };
+async function sendAnonymousMessage(channelId: string, message: string, replyMessageId?: string) {
+  let payload: SendPayloadType = {channelId, message};
+  if (replyMessageId) {
+    payload = {...payload, messageType: 'reply', replyMessageId};
+  }
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `sendAnonymousMessage` function has been updated to include an optional `replyMessageId` parameter. This is a good change as it allows for reply messages to be sent. However, the `messageType` is hardcoded as 'reply' when `replyMessageId` is present. It would be better to make this more flexible by allowing `messageType` to be passed as a parameter. This way, if more message types are added in the future, the function won't need to be modified.

```diff
-  let payload: SendPayloadType = {channelId, message};
-  if (replyMessageId) {
-    payload = {...payload, messageType: 'reply', replyMessageId};
-  }
+  let payload: SendPayloadType = {channelId, message, messageType, replyMessageId};
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->